src/asm/lexer.c

```diff
@@ -921,6 +921,31 @@ void lexer_DumpStringExpansions(void)
 	free(stack);
 }
 
+/* Discards an inline comment */
+static void discardInlineComment(void)
+{
+	dbgPrint("Discarding inline comment\n");
+	int c;
+	shiftChars(2);
+	for (;;) {
+		c = peek(0);
+		if (c == EOF) {
+			error("Unterminated inline comment\n");
+			return;
+		}
+		if (c == '/' && peek(1) == '*') {
+			error("Inline comment contents can't contain /*\n");
```

Comment:
This should rather be a warning; it might be legitimate to have these characters in a comment (example: `/* Handle binary ops +/* now */`), and it can still be made into an error if desired (using `-Werror=xxx`). The question then becomes which warning flag to use; I'd suggest `-Wnested-comment`.

Reply:
warning is pushed up, made the change before seeing your note so it's the verbose `-Wcomment-in-block-comment`. i can update to `-Wnested-comment`, but it's different than `; comment ; comment`, so it might be clearer to call it `block` in the warning

Reply:
The rationale for that warning is that a block comment starter within a block comment may indicate a runaway block comment (forgot to close it, only closed by a later comment).

Line comments, however, have much more limited scope, and thus cannot really be made runaway :P